src/index.ts

```diff
@@ -12,13 +22,47 @@ type HeadlessConfig = AnswersConfig;
  *                 experience.
  */
 export function provideAnswersHeadless(config: HeadlessConfig): AnswersHeadless {
-  const answersCore = provideCore(config);
-  const stateManager = new ReduxStateManager();
+  const {
+    headlessId,
+    ...answersConfig
+  } = config;
+  if (headlessId === DEFAULT_HEADLESS_ID) {
+    throw new Error(`Cannot instantiate an AnswersHeadless using the default headlessId "${headlessId}". `
+      + 'Specify a different headlessId.');
+  }
+  const answersCore = provideCore(answersConfig);
+  const stateManager = new ReduxStateManager(
+    store, headlessId || DEFAULT_HEADLESS_ID, headlessReducerManager);
   const httpManager = new HttpManager();
 
-  return new AnswersHeadless(answersCore, stateManager, httpManager);
+  const headless = new AnswersHeadless(answersCore, stateManager, httpManager);
+  if (!firstHeadlessInstance) {
+    firstHeadlessInstance = headless;
+  } else {
+    // Two-way bind the current headless instances with the first one instantiated on the page.
+    // This way, all headless instances on a page will have their sessionTracking states linked.
+    // We have to be careful not to create an infinite loop here.
```

Comment:
Are we avoiding the infinite loop because the listener is only fired when the value of the valueAccessor changes?

Reply:
Yup! We have to be careful about reference equality.
let's say we have a `first` and `second` headless. The flow looks like this
1. we call `first.setSessionId('blah')`
2. this triggers the listener to `first`'s sessionTracking, which calls `second.setState({...second.state, sessionTracking })`
4. because `second`'s sessionTracking has been updated, `second`'s sessionTracking listener calls `first.setState({...first.state, sessionTracking })`
5. We come back to `first`'s listener. Here, because we were careful about reference equality for sessionTracking, the listener sees that the reference has not changed, and the loop ends.

It's important to note that if we used the spread operator in the setState like 
```ts
first.setState({...first.state, sessionTracking: { ...sessionTracking } )
```
We would run into an infinite loop, or if the listener listened to the entire state instead of just `state.sessionTracking`

Reply:
I see, we avoid the infinite loop since `sessionTracking` is referencing the same object in state